pkg/migrations/sql/idempotent/base.sql

```diff
@@ -1983,6 +2031,71 @@ SET search_path = pg_temp;
 REVOKE ALL ON FUNCTION SCHEMA_CATALOG.create_metric_view(text) FROM PUBLIC;
 GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.create_metric_view(text) TO prom_writer;
 
+CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.register_metric_view(schema_name name, view_name name, if_not_exists BOOLEAN = false)
```

Comment:
we should also add an unregister function

Reply:
also we should add a test that we register a view and then drop it without registering. Subsequent queries should error. But, that error should be nice and clear.